runtime/js/40_command.js

```diff
@@ -0,0 +1,168 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+  const { assert, pathFromURL } = window.__bootstrap.util;
+  const { read } = window.__bootstrap.io;
+  const { writeAll } = window.__bootstrap.buffer;
+  const { illegalConstructorKey } = window.__bootstrap.webUtil;
+  const { ArrayPrototypeMap, ObjectEntries, String } =
+    window.__bootstrap.primordials;
+
+  class Command {
+    #args;
+
+    constructor(command, {
+      args = [],
+      cwd = undefined,
+      clearEnv = false,
+      env = {},
+    } = {}) {
+      this.#args = {
+        cmd: pathFromURL(command),
+        args: ArrayPrototypeMap(args, String),
+        cwd: pathFromURL(cwd),
+        clearEnv,
+        env: ObjectEntries(env),
+      };
+    }
+
+    spawn(options = {}) {
+      const child = core.opSync("op_command_spawn", {
+        ...this.#args,
+        ...options,
+      });
+
+      return new Child(illegalConstructorKey, child);
+    }
+
+    async status(options = {}) {
+      return await core.opAsync("op_command_status", {
+        ...this.#args,
+        ...options,
+      });
+    }
+
+    async output() {
+      return await core.opAsync("op_command_output", this.#args);
+    }
+  }
+
+  class Child {
+    #rid;
+
+    #pid;
+    get pid() {
+      return this.#pid;
+    }
+
+    #stdin;
+    get stdin() {
+      return this.#stdin;
+    }
+
+    #stdoutRid;
+    #stdout;
+    get stdout() {
+      return this.#stdout;
+    }
+
+    #stderrRid;
+    #stderr;
+    get stderr() {
+      return this.#stderr;
+    }
+
+    constructor(key = null, {
+      rid,
+      pid,
+      stdinRid,
+      stdoutRid,
+      stderrRid,
+    } = null) {
+      if (key !== illegalConstructorKey) {
+        throw new TypeError("Illegal constructor.");
+      }
+
+      this.#rid = rid;
+      this.#pid = pid;
+
+      if (stdinRid !== undefined) {
+        this.#stdin = new WritableStream({
```

Comment:
One thing about using WHATWG streams is that we can support piping to stdout/stderr from any other WHATWG stream if that was implemented in JS instead of Rust (tangential, but ref #2750)